File: Python/statsig.py
import os
import getpass
from statsig_python_core import Statsig, StatsigOptions, StatsigUser
from dotenv import load_dotenv

options = StatsigOptions()
options.environment = "development"

user = os.environ.get("USER") or getpass.getuser()
print(user)
statsigUser = StatsigUser(user)

load_dotenv()  # Loads variables from .env into environment

statsig = Statsig(os.getenv("SERVER_KEY"), options)
statsig.initialize().wait()

gatePass = statsig.check_gate(statsigUser, "new_feature_gate")
if(gatePass):
    print("The new feature is enabled!")
else:
    print("The new feature is disabled.")

statsig.log_event( 
    user=statsigUser,
    event_name="check_gate_passed",
    value=gatePass,
    metadata={},
)

statsig.shutdown().wait()

File: Go/statsig.go
package main

import (
	"fmt"

	"os"

	statsig "github.com/statsig-io/go-sdk"

	godotenv "github.com/joho/godotenv"
)

func main() {
	// Load .env from parent directory
	godotenv.Load("../.env")
	statsig.InitializeWithOptions(os.Getenv("SERVER_KEY"), &statsig.Options{Environment: statsig.Environment{Tier: "development"}})

	// Get the logged-in user's ID from environment variable
	userID := os.Getenv("USER")
	fmt.Println("User ID: " + userID)
	user := statsig.User{UserID: userID}
	feature := statsig.CheckGate(user, "new_feature_gate")

	status := "disabled"
	if feature {
		status = "enabled"
	}
	fmt.Println("Feature is " + status)

	statsig.LogEvent(statsig.Event{
		User: user,
		EventName: "app_started",
                Value: status,
		Metadata: nil,
	})

	// Shutdown to ensure all events are flushed before exiting
	statsig.Shutdown()
}